elliottlib/cli/find_bugs_kernel_cli.py

```diff
@@ -0,0 +1,335 @@
+import re
+import sys
+from typing import Any, Dict, List, Optional, Sequence, TextIO, Tuple, cast
+
+import click
+import koji
+from bugzilla import Bugzilla
+from bugzilla.bug import Bug
+from jira import JIRA, Issue
+
+from elliottlib import Runtime, brew
+from elliottlib.assembly import AssemblyTypes
+from elliottlib.cli.common import cli, click_coroutine
+from elliottlib.config_model import KernelBugSweepConfig
+from elliottlib.exceptions import ElliottFatalError
+from elliottlib.util import green_print
+
+
+class FindBugsKernelCli:
+    def __init__(self, runtime: Runtime, trackers: Sequence[str],
+                 clone: bool, reconcile: bool, comment: bool, dry_run: bool):
+        self._runtime = runtime
+        self._logger = runtime.logger
+        self.trackers = list(trackers)
+        self.clone = clone
+        self.reconcile = reconcile
+        self.comment = comment
+        self.dry_run = dry_run
+        self._id_bugs: Dict[int, Bug] = {}  # cache for kernel bug; key is bug_id, value is Bug object
+        self._bug_trackers: Dict[int, Issue] = {}  # bug_id -> KMAINT jira mapping
+
+    async def run(self):
+        logger = self._logger
+        if self.reconcile and not self.clone:
+            raise ElliottFatalError("--reconcile must be used with --clone")
+        if self._runtime.assembly_type is not AssemblyTypes.STREAM:
+            raise ElliottFatalError("This command only supports stream assembly")
+        group_config = self._runtime.group_config
+        raw_config = self._runtime.gitdata.load_data(key='bug', replace_vars=group_config.vars).data.get("kernel_bug_sweep")
+        if not raw_config:
+            logger.warning("kernel_bug_sweep is not defined in bug.yml")
+            return
+        config = KernelBugSweepConfig.parse_obj(raw_config)
+        jira_tracker = self._runtime.bug_trackers("jira")
+        jira_client: JIRA = jira_tracker._client
+        bz_tracker = self._runtime.bug_trackers("bugzilla")
+        bz_client: Bugzilla = bz_tracker._client
+        koji_api = self._runtime.build_retrying_koji_client(caching=True)
+
+        # Getting KMAINT trackers
+        trackers_keys = self.trackers
+        trackers: List[Issue] = []
+        if not trackers_keys:
+            logger.info("Searching for open trackers...")
+            trackers = self._find_kmaint_trackers(jira_client, config.tracker_jira.project, config.tracker_jira.labels)
```

Comment:
Initially my thought was that this is not a functionality that is shared between Jira and Bugzilla, so I didn't put the logic there. But I am ok to move this to jira bug tracker if you think it is better.

Reply:
@thegreyd I found a lot of methods in JIRABugTracker filters on project (OCPBUGS) and accesses to fields specific to OCPBUGS. It might require some changes to the existing implementation to make it work for other projects. TBH I don't really want to make a lot of changes to JIRABugTracker since it is critical part to a lot of commands of elliott. Let me know if you have any ideas.